tests/app/authentication/test_authentication.py

```diff
@@ -424,6 +424,11 @@ def endpoint_that_requires_user_in_service():
 
         user = create_user()
         service = create_service(service_name='some-service')
+        dao_add_user_to_service(
+            service,
+            user,
+            permissions=[Permission(service=service, user=user, permission='manage_users')]
+        )
```

Comment:
I would add a comment about why this is necessary.  It isn't obvious to me just looking at the code.  I know it addresses a test failure.

Reply:
I am not sure why it has been failed with my changes, not in existing changes. But `requires_user_in_service_or_admin` middleware verify user permissions assign to a service so we have to add user into the mock service.

Reply:
done